c++/include/orc/Reader.hh

```diff
@@ -605,6 +612,21 @@ namespace orc {
      */
     virtual std::map<uint32_t, BloomFilterIndex> getBloomFilters(
         uint32_t stripeIndex, const std::set<uint32_t>& included) const = 0;
+
+    /**
+     * Get the input stream for the ORC file.
+     */
+    virtual InputStream* getStream() const = 0;
```

Comment:
I don't think we need to add it to the base class. You can cast to `ReaderImpl` when required.